sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -185,10 +185,17 @@ export class Receiver {
         ? String(lockTokenOrMessage.lockToken)
         : String(lockTokenOrMessage);
 
-    const lockedUntilUtc = await this._context.managementClient!.renewLock(lockToken);
-    if (lockTokenOrMessage instanceof ServiceBusMessage) {
-      lockTokenOrMessage.lockedUntilUtc = lockedUntilUtc;
+    let receiverName;
+    if (this._context.batchingReceiver) {
+      receiverName = BatchingReceiver.name;
+    } else if (this._context.streamingReceiver) {
+      receiverName = this._context.streamingReceiver.name;
```

Comment:
Take the case when user first calls `receiveMessages` and then `registerMessageHandler`. At this point, both`this._context.batchingReceiver` and `this._context.streamingReceiver.name` will exist.

How do we make sure we pass the right receiver name? Does it even matter that we pass the right receiver name.

Reply:
That's an interesting but different scenario. 
It may surface a bugger design problem, just added the test. Letting it run for 20 min to see what happens.

Reply:
If the test fails, then please log a new issue.